src/Trace/TraceClient.php

```diff
@@ -144,74 +151,37 @@ public function insertBatch(array $traces, array $options = [])
+        return new Trace($this->projectId, $traceId);
     }
 
-    /**
-     * Fetch all traces in the project.
-     *
-     * @see https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces/list Traces list API documentation.
-     *
-     * @param array $options [optional] {
-     *      Configuration options.
-     *
-     *      @type string $viewType Type of data returned for traces in the list.
-     *            Can be one of 'VIEW_TYPE_UNSPECIFIED', 'MINIMAL', 'ROOTSPAN', or
-     *            'COMPLETE'
-     *      @type int $pageSize Maximum number of traces to return per page.
-     *      @type int $resultLimit Limit the number of results returned in total.
-     *           **Defaults to** `0` (return all results).
-     *      @type string $pageToken Token identifying the page of results to return
-     *      @type string $startTime Start of the time interval during which trace data
-     *            was collected. This timestamp in nanoseconds should be in "Zulu" format.
-     *            Example: '2014-10-02T15:01:23.045123456Z'
-     *      @type string $endTime End of the time interval during which trace data was
-     *            collected. This timestamp in nanoseconds should be in "Zulu" format.
-     *            Example: '2014-10-02T15:01:23.045123456Z'
-     *      @type string $filter An optional filter for the request
-     *      @type string $orderBy Field used to sort the returned traces. Can be one
-     *            of 'traceId', 'name', 'duration', 'start'. Descending order can be
-     *            specified by appending 'desc' to the sort field (for example,
-     *            'name desc'). Only one sort field is permitted.
-     * }
-     * @return ItemIterator<Trace>
-     */
-    public function traces(array $options = [])
-    {
-        $resultLimit = $this->pluck('resultLimit', $options, false);
-
-        return new ItemIterator(
-            new PageIterator(
-                function (array $trace) {
-                    $trace += ['spans' => null];
-                    return new Trace($this->connection, $trace['projectId'], $trace['traceId'], $trace['spans']);
-                },
-                [$this->connection, 'listTraces'],
-                ['projectId' => $this->projectId] + $options,
-                [
-                    'itemsKey' => 'traces',
-                    'resultLimit' => $resultLimit
-                ]
-            )
-        );
-    }
-
-    /**
-     * Return a Trace reporter that utilizes this client's configuration
-     *
-     * @param  array $options [optional] Reporter options.
-     *      {@see Google\Cloud\Trace\Reporter\AsyncReporter::__construct()}
-     * @return ReporterInterface
-     */
-    public function reporter(array $options = [])
+    private function transformSpan($trace)
     {
-        return new AsyncReporter($options + [
-            'clientConfig' => $this->clientConfig
-        ]);
+        $projectId = $this->projectId;
```

Comment:
You should be able to reference `$this->projectId` in the anonymous function and not have to do this.